tests/system/src/test/java/test/javafx/embed/swing/JFXPanelNPETest.java

```diff
@@ -0,0 +1,126 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.embed.swing;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.Scene;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+
+public class JFXPanelNPETest {
+    private static WebView webView;
+    private static JFrame jFrame;
+    private static JFXPanel contentPane;
+    private static AtomicBoolean failure;
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Application class. An instance is created and initialized before running
+    // the first test, and it lives through the execution of all tests.
+    public static class MyApp extends Application {
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            Platform.setImplicitExit(false);
+            Assert.assertTrue(Platform.isFxApplicationThread());
+            Assert.assertNotNull(primaryStage);
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void doSetupOnce() throws Exception {
+        Util.launch(launchLatch, MyApp.class);
+        Assert.assertEquals(0, launchLatch.getCount());
+    }
+
+    @AfterClass
+    public static void doTeardownOnce() {
+        Util.shutdown();
+    }
+
+    @After
+    public void doCleanup() {
+        if (jFrame != null) {
+            SwingUtilities.invokeLater(() -> jFrame.dispose());
+        }
+    }
+
+    @Test
+    public void testSceneNPE() throws Exception {
+        failure = new AtomicBoolean(false);
+        Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                e.printStackTrace();
+                failure.set(true);
+            }
+        });
+        SwingUtilities.invokeAndWait(JFXPanelNPETest::createUI);
+        for (int i = 0; i < 300; i++) {
+            SwingUtilities.invokeLater(contentPane::repaint);
+            Platform.runLater(() -> contentPane.setScene(null));
+            Thread.sleep(100);
```

Comment:
I would recommend replacing a fixed number with a random value.

(Typically, when a random is added to the test, it would make sense to print its seed to stdout, for the sake of reproducing a possible failure later; but in this case it might be unnecessary because we are dealing with multiple threads and that adds its own degree of randomness, completely eliminating reproducibility.  Or, perhaps, we still need to print the seed to maximize the probability of reproducing the failed scenario.)

Reply:
Despite downvote, I still think randomness in the unit test is a good thing.

For more background, please take a look at the second answer in 
https://stackoverflow.com/questions/32458/random-data-in-unit-tests